plugins/modules/system/alternatives.py

```diff
@@ -141,18 +184,34 @@ def main():
                     [UPDATE_ALTERNATIVES, '--install', link, name, path, str(priority)],
                     check_rc=True
                 )
+                changed = True
 
-            # select the requested path
+            # set the current selection to this path (if requested)
+            if state == AlternativeState.SELECTED:
+                module.run_command(
+                    [UPDATE_ALTERNATIVES, '--set', name, path],
+                    check_rc=True
+                )
+                changed = True
+
+        except subprocess.CalledProcessError as cpe:
```

Comment:
Is this not covered by `check_rc=True`?

Reply:
Hmm, actually `subprocess.CalledProcessError` should never leave `module.run_command()`. The module has been catching it before, I guess both the old `catch` and the new one can be removed.

Reply:
That can also be done later.